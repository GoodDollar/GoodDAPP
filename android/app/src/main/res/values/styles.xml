<resources>

    <!-- Base application theme. -->
    <style name="AppTheme" parent="Theme.AppCompat.Light.NoActionBar">
        <!-- Customize your theme here. -->
        <item name="android:textColor">#000000</item>
        <item name="android:forceDarkAllowed">false</item>
    </style>

    <style name="SplashTheme" parent="Theme.AppCompat.DayNight.NoActionBar">
        <item name="android:windowBackground">@drawable/splash</item>
    </style>


</resources>
