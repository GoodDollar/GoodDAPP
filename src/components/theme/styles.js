import DefaultTheme from 'react-native-paper/src/styles/DefaultTheme'
import normalize from '../../lib/utils/normalizeText'

export const theme = {
  ...DefaultTheme,
  borders: {
    defaultBorderColor: '#c9c8c9',
  },
  colors: {
    ...DefaultTheme.colors,
    facebookBlue: '#1877F3',
    googleBlue: '#3367D6',
    blue: '#006EA0',
    darkBlue: '#0C263D',
    darkGray: '#42454A',
    gray: '#555',
    gray50Percent: '#CBCBCB',
    gray80Percent: '#A3A3A3',
    placeholder: '#CBCBCB',
    disabled: '#E3E3E2',
    green: '#00C3AE',
    lightGreen: '#00BBA5',
    lightGray: '#EEE',
    lighterGray: '#A3A3A3',
    orange: '#F8AF40',
    primary: '#00AFFF',
    purple: '#9F6A9D',
    red: '#FA6C77',
    googleRed: '#EA4335',
    error: '#FA6C77',
    text: '#222',
    white: '#ffffff',
  },
  fonts: {
    ...DefaultTheme.fonts,
    default: 'Roboto',
    slab: 'Roboto Slab',
  },
  paddings: {
    defaultMargin: 8,
    mainContainerPadding: 8,
  },
  sizes: {
    default: 8,
    defaultDouble: 16,
    defaultQuadruple: 32,
    defaultHalf: 4,
    borderRadius: 5,
    maxHeightForTabletAndDesktop: 844,
    minHeightForDialogMessage: 350,
    maxWidthForTabletAndDesktop: 475,
  },
  fontStyle: {
    color: '#555',
    fontSize: normalize(18),
    textAlign: 'center',
  },
  backdrop: {
    backgroundColor: 'rgba(255, 0, 0, 0.8)',
  },
  modals: {
    overlayBackgroundColor: 'transparent',
    activityIndicatorBackgroundColor: 'rgba(255, 255, 255, 0.8)',
    overlayHorizontalPadding: 20,
    overlayVerticalPadding: 65,
    backgroundColor: '#fff',
    jaggedEdgeSize: 16,
    contentPadding: 16,
    borderRadius: 5,
    borderLeftWidth: 10,
  },
  feedItems: {
    borderRadius: 8,
    itemBackgroundColor: '#fff',
    height: normalize(84),
  },
}
