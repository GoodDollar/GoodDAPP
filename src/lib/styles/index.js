import React from 'react'
import { useTheme, withTheme } from 'react-native-paper'
import { StyleSheet } from 'react-native'

import { isFunction } from 'lodash'

export const withOpacity = (color, opacity) => {
  let decimal = opacity

  if (!Number.isInteger(opacity) && opacity <= 1) {
    decimal = Math.round(255 * opacity)
  }

  return `${color}${decimal.toString(16)}`
}

/**
 * HOC that injects `theme` from `withTheme` and `styles` using theme values into the `Component`
 * @param {*} mapThemeToStyles receives props and retuns an object with calculated styles
 * @param {*} withStyleSheet wheather should or shouldn't be the result wrapped with `StyleSheet.create`
 */
export const withStyles = (mapThemeToStyles, withStyleSheet = true) => Component => {
  const getUpdatedStyles = props => {
    let styles = {}
    if (isFunction(mapThemeToStyles)) {
      const stylesObject = mapThemeToStyles(props)
      styles = withStyleSheet ? StyleSheet.create(stylesObject) : stylesObject
    }
    return styles
  }

  class WrappedComponent extends React.Component {
    constructor(props) {
      super(props)
      const styles = getUpdatedStyles(props)
      this.state = { styles }
    }

    static getDerivedStateFromProps(props, state) {
      const styles = getUpdatedStyles(props)
      return { styles }
    }

    _root

    /**
     * @internal
     * Required for when the component is wrapped inside a TouchableOpacity* element
     * https://facebook.github.io/react-native/docs/direct-manipulation
     */
    setNativeProps(...args) {
      return this._root && this._root.setNativeProps(...args)
    }

    render() {
      if (!isFunction(mapThemeToStyles)) {
        return <Component ref={c => (this._root = c)} {...this.props} />
      }

      return <Component {...this.props} styles={this.state.styles} />
    }
  }

  // Adding static methods to new wrapped component
  Object.getOwnPropertyNames(Component)
    .filter(prop => isFunction(Component[prop]))
    .forEach(staticMethod => (WrappedComponent[staticMethod] = Component[staticMethod]))
  WrappedComponent.navigationOptions = Component.navigationOptions

  return withTheme(WrappedComponent)
}

export const makeStyles = (mapThemeToStyles, withStyleSheet = true) => {
  const getUpdatedStyles = props => {
    let styles = {}

    if (isFunction(mapThemeToStyles)) {
      const stylesObject = mapThemeToStyles(props)

      styles = withStyleSheet ? StyleSheet.create(stylesObject) : stylesObject
    }

    return styles
  }

  const useStyles = props => {
    const theme = useTheme()

    return { theme, styles: getUpdatedStyles({ ...props, theme }) }
  }

  return useStyles
}
