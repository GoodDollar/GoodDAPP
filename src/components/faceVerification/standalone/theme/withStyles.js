// utils
import { getDesignRelativeHeight, getDesignRelativeWidth, isLargeDevice } from '../../../../lib/utils/sizes'
import normalize from '../../../../lib/utils/normalizeText'
import { withStyles } from '../../../../lib/styles'
import { isBrowser } from '../../../../lib/utils/platform'

const getStylesFromProps = ({ theme }) => ({
  topContainer: {
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: theme.colors.surface,
    borderRadius: theme.sizes.borderRadius,
    display: 'flex',
    flexDirection: 'column',
    flexShrink: 0,
    paddingBottom: getDesignRelativeHeight(theme.sizes.defaultDouble),
    paddingLeft: getDesignRelativeWidth(theme.sizes.default),
    paddingRight: getDesignRelativeWidth(theme.sizes.default),
    paddingTop: getDesignRelativeHeight(theme.sizes.defaultDouble),
  },

  mainContent: {
    flexGrow: 1,
    justifyContent: 'center',
    width: '100%',
  },

  descriptionContainer: {
    paddingHorizontal: getDesignRelativeHeight(theme.sizes.defaultHalf),
    paddingVertical: getDesignRelativeHeight(isBrowser ? theme.sizes.defaultDouble : 14),
    alignItems: 'center',
  },
  descriptionContainerB: {
    paddingVertical: getDesignRelativeHeight(isBrowser ? 12 : 10),
  },
  descriptionWrapper: {
    flexDirection: 'column',
    alignItems: 'center',
  },

  text: {
    textAlign: 'center',
    fontSize: normalize(isLargeDevice ? 22 : 20),
    lineHeight: isLargeDevice ? 36 : 34,
  },

  infoRow: {
    justifyContent: 'flex-start',
    alignItems: 'center',
  },

  action: {
    width: '100%',
  },
  actionsSpace: {
    marginBottom: getDesignRelativeHeight(16),
  },
})

export default withStyles(getStylesFromProps)
