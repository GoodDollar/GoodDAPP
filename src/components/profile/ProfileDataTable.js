import React, { Fragment, useCallback, useMemo } from 'react'
import { Platform, StyleSheet, View } from 'react-native'
import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view'
import { get, noop } from 'lodash'
import { parsePhoneNumberFromString } from 'libphonenumber-js'
import { t } from '@lingui/macro'
import useCountryFlag from '../../lib/hooks/useCountryFlag'
import Icon from '../common/view/Icon'
import InputRounded from '../common/form/InputRounded'
import ErrorText from '../common/form/ErrorText'
import Section from '../common/layout/Section'
import { withStyles } from '../../lib/styles'
import API from '../../lib/API'
import PhoneInput from './PhoneNumberInput/PhoneNumberInput'

const defaultErrors = {}
const defaultStoredProfile = {}
const defaultProfile = {}

export const CountryFlag = withStyles(
  () => ({
    flag: {
      width: 30,
      height: 30,
    },
  }),
  false,
)(({ styles, code }) => {
  if (!code) {
    return null
  }

  const Flag = useCountryFlag(code)

  if (!Flag) {
    return null
  }

  return Flag
})

const ProfileDataTable = ({
  profile = defaultProfile,
  storedProfile = defaultStoredProfile,
  onChange,
  errors = defaultErrors,
  editable,
  theme,
  styles,
  screenProps,
  setLockSubmit = noop,
  showCustomFlag,
}) => {
  const { mobile } = profile || {}
  const phoneMeta = useMemo(() => (showCustomFlag && mobile ? parsePhoneNumberFromString(mobile) : null), [
    showCustomFlag,
    mobile,
  ])
  const { country: countryCode = null } = phoneMeta || {}

  const verifyEdit = useCallback(
    (field, content) => {
      screenProps.push('VerifyEdit', { field, content })
    },
    [screenProps],
  )

  const verifyEmail = useCallback(() => {
    if (profile.email !== storedProfile.email) {
      verifyEdit('email', profile.email)
    }
  }, [verifyEdit, profile.email, storedProfile.email])

  const verifyPhone = useCallback(async () => {
    if (!storedProfile.mobile) {
      const onlyCheckAlreadyVerified = true
      const res = await API.sendOTP({ mobile }, onlyCheckAlreadyVerified)

      if (!get(res, 'data.alreadyVerified', false)) {
        verifyEdit('phone', mobile)
      }
    } else if (mobile !== storedProfile.mobile) {
      verifyEdit('phone', mobile)
    }
  }, [verifyEdit, mobile, storedProfile.mobile])

  // username handlers
  const onUserNameChange = useCallback(username => onChange(profile.update({ username })), [onChange, profile])

  // phone handlers
  const onPhoneInputFocus = useCallback(() => setLockSubmit(true), [setLockSubmit])
  const onPhoneInputChange = useCallback(value => onChange(profile.update({ mobile: value })), [onChange, profile])
  const onPhoneInputBlur = useCallback(() => {
    const { errors: _errors } = profile.validate()
    const isValid = !_errors.mobile

    setLockSubmit(!isValid)

    if (isValid && mobile) {
      verifyPhone()
    }
  }, [setLockSubmit, verifyPhone, errors, onChange, profile])
  const phoneInputStyles = useMemo(() => StyleSheet.flatten(styles.phoneInput), [styles.phoneInput])

  // email handlers
  const onEmailFocus = useCallback(() => setLockSubmit(true), [setLockSubmit])
  const onEmailChange = useCallback(email => onChange(profile.update({ email })), [onChange, profile])
  const onEmailBlur = useCallback(() => {
    const { errors: _errors } = profile.validate()

    if (!_errors.email) {
      setLockSubmit(false)
      verifyEmail()
    }
  }, [setLockSubmit, verifyEmail, errors])

  return (
    <Section.Row alignItems="center" grow={1}>
      <KeyboardAwareScrollView resetScrollToCoords={{ x: 0, y: 0 }} scrollEnabled={false}>
        <Section.Row>
          <InputRounded
            disabled={!editable}
            error={errors.username}
            icon="username"
            iconColor={theme.colors.primary}
            iconSize={22}
            onChange={onUserNameChange}
            placeholder={t`Choose a Username`}
            value={profile.username}
          />
        </Section.Row>
        <Section.Row>
          {editable ? (
            <Section.Stack grow>
              <Section.Row style={phoneInputStyles}>
                <PhoneInput
                  error={!!errors.mobile}
                  id="signup_phone"
                  onFocus={onPhoneInputFocus}
                  onChange={onPhoneInputChange}
                  onBlur={onPhoneInputBlur}
                  placeholder="Enter phone number"
                  value={mobile}
                  textStyle={{ color: errors.mobile && theme.colors.red }}
                />
                <Section.Row style={styles.suffixIcon}>
                  <Icon
                    color={errors.mobile ? theme.colors.red : theme.colors.primary}
                    name="phone"
                    size={28}
                    style={styles.phoneIcon}
                  />
                </Section.Row>
              </Section.Row>
              <ErrorText error={errors.mobile} style={styles.errorMargin} />
            </Section.Stack>
          ) : (
            <Fragment>
              {countryCode && (
                <View style={styles.flagContainer}>
                  <CountryFlag code={countryCode} />
                </View>
              )}
              <InputRounded
                containerStyle={countryCode && styles.disabledPhoneContainer}
                disabled={true}
                error={errors.mobile}
                icon="phone"
                iconColor={theme.colors.primary}
                iconSize={28}
                placeholder="Add your Mobile"
                value={mobile ?? ''}
              />
            </Fragment>
          )}
        </Section.Row>
        <Section.Row style={!editable && styles.borderedBottomStyle}>
          <InputRounded
            disabled={!editable}
            error={errors.email}
            icon="envelope"
            iconColor={theme.colors.primary}
            iconSize={20}
            onFocus={onEmailFocus}
            onChange={onEmailChange}
            onBlur={onEmailBlur}
            placeholder="Add your Email"
            value={profile.email ?? ''}
          />
        </Section.Row>
      </KeyboardAwareScrollView>
    </Section.Row>
  )
}

const getStylesFromProps = ({ theme, errors }) => {
  return {
    borderedBottomStyle: {
      borderBottomColor: theme.colors.lightGray,
      borderBottomWidth: 1,
    },
    suffixIcon: {
      alignItems: 'center',
      display: 'flex',
      height: 40,
      justifyContent: 'center',
      position: 'absolute',
      right: 1,
      width: 32,
      zIndex: 1,
    },
    errorMargin: {
      marginTop: theme.sizes.default,
      marginBottom: theme.sizes.default,
    },
    phoneIcon: {
      paddingTop: Platform.select({
        web: 0,
        default: 5,
      }),
    },
    phoneInput: {
      borderColor: errors && (errors.mobile ? theme.colors.red : theme.colors.lightGray),
      borderRadius: 24,
      borderWidth: 1,
      color: errors && (errors.mobile ? theme.colors.red : theme.colors.text),
      position: 'relative',
      marginVertical: 4,
      paddingLeft: 10,
      paddingRight: 10,
      paddingTop: 6,
      paddingBottom: 6,
    },
    flagContainer: {
      height: 24,
      width: 24,
      justifyContent: 'center',
      alignItems: 'center',
      overflow: 'hidden',
      borderWidth: 1,
      borderColor: theme.colors.lightGray,
      ...Platform.select({
        web: { borderRadius: '50%', borderStyle: 'solid' },
        default: { borderRadius: 24 / 2 },
      }),
    },
    disabledPhoneContainer: {
      paddingLeft: 10,
    },
  }
}

export default withStyles(getStylesFromProps)(ProfileDataTable)
