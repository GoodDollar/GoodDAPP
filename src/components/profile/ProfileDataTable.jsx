import React, { Fragment, useCallback, useMemo } from 'react'
import { Platform, StyleSheet, Text, View } from 'react-native'
import { KeyboardAwareScrollView } from '@mtourj/react-native-keyboard-aware-scroll-view'
import { get, noop } from 'lodash'
import { parsePhoneNumberFromString } from 'libphonenumber-js'
import { t } from '@lingui/macro'

import useCountryFlag from '../../lib/hooks/useCountryFlag'
import Icon from '../common/view/Icon'
import InputRounded from '../common/form/InputRounded'
import { getDesignRelativeHeight, getDesignRelativeWidth } from '../../lib/utils/sizes'
import ErrorText from '../common/form/ErrorText'
import Section from '../common/layout/Section'
import { withStyles } from '../../lib/styles'
import API from '../../lib/API'
import { SvgXml } from '../common'
import PhoneInput from './PhoneNumberInput/PhoneNumberInput'
import VerifyButton from './VerifyButton'

const defaultErrors = {}
const defaultStoredProfile = {}
const defaultProfile = {}

const ProfileDataTable = ({
  profile = defaultProfile,
  storedProfile = defaultStoredProfile,
  onChange,
  errors = defaultErrors,
  editable,
  theme,
  styles,
  screenProps,
  setSaveMode = noop,
  saveMode,
  showCustomFlag,
  editMode = false,
}) => {
  const { mobile } = profile || {}
  const phoneMeta = useMemo(
    () => (showCustomFlag && mobile ? parsePhoneNumberFromString(mobile) : null),
    [showCustomFlag, mobile],
  )
  const { country: countryCode = null } = phoneMeta || {}
  const countryFlag = useCountryFlag(countryCode)

  const verifyEdit = useCallback(
    (field, content) => {
      screenProps.push('VerifyEdit', { field, content })
    },
    [screenProps],
  )

  const verifyEmail = useCallback(() => {
    if (profile.email !== storedProfile.email) {
      verifyEdit('email', profile.email)
    }
  }, [verifyEdit, profile.email, storedProfile.email])

  const verifyPhone = useCallback(async () => {
    if (!storedProfile.mobile) {
      const onlyCheckAlreadyVerified = true
      const res = await API.sendOTP({ mobile }, onlyCheckAlreadyVerified)

      if (!get(res, 'data.alreadyVerified', false)) {
        verifyEdit('phone', mobile)
      }
    } else if (mobile !== storedProfile.mobile) {
      verifyEdit('phone', mobile)
    }
  }, [verifyEdit, mobile, storedProfile.mobile])

  // username handlers
  const onUserNameChange = useCallback(
    fullName => {
      onChange(profile.update({ fullName }), 'name')
      setSaveMode(prev => ({ ...prev, name: true }))
    },
    [onChange, profile],
  )

  // phone handlers
  const onPhoneInputChange = useCallback(
    value => {
      onChange(profile.update({ mobile: value }), 'mobile')
      setSaveMode(prev => ({ ...prev, phone: true }))
    },
    [setSaveMode, profile],
  )
  const onVerifyPhone = useCallback(() => {
    const { errors: _errors } = profile.validate()
    const isValid = !_errors.mobile

    if (isValid && mobile) {
      verifyPhone()
    }
  }, [verifyPhone, errors, onChange, profile])
  const phoneInputStyles = useMemo(() => StyleSheet.flatten(styles.phoneInput), [styles.phoneInput])

  // email handlers
  const onEmailChange = useCallback(
    email => {
      onChange(profile.update({ email }), 'email')
      setSaveMode(prev => ({ ...prev, email: true }))
    },
    [setSaveMode, profile],
  )

  const onVerifyEmail = useCallback(() => {
    const { errors: _errors } = profile.validate()

    if (!_errors.email) {
      verifyEmail()
    }
  }, [verifyEmail, errors])

  return (
    <Section.Row alignItems={editMode ? 'flex-start' : 'center'} grow={1}>
      <KeyboardAwareScrollView resetScrollToCoords={{ x: 0, y: 0 }} scrollEnabled={false}>
        {editMode && (
          <Section.Row style={[styles.disclaimer, { opacity: 0.8 }]}>
            <Text style={{ fontFamily: 'Roboto', fontSize: 14 }}>
              <Text style={{ fontWeight: 'bold' }}>{t`Note:`} </Text>
              {t`Changing your information here will not change how you log in to your wallet.`}
            </Text>
          </Section.Row>
        )}
        {editMode && (
          <Section.Row>
            <InputRounded
              disabled={!editable}
              error={errors.username}
              icon="username"
              iconColor={theme.colors.primary}
              iconSize={22}
              onChange={onUserNameChange}
              value={profile.fullName}
            />
          </Section.Row>
        )}
        <Section.Row>
          {editable ? (
            <Section.Stack grow>
              <Section.Row style={phoneInputStyles}>
                <PhoneInput
                  error={!!errors.mobile}
                  id="signup_phone"
                  onChange={onPhoneInputChange}
                  placeholder="Enter phone number"
                  value={mobile}
                  textStyle={{ color: errors.mobile && theme.colors.red }}
                />
                {saveMode?.phone ? (
                  <Section.Row styles={styles.customSuffixIcon}>
                    <VerifyButton enabled={true} mode="phone" cb={onVerifyPhone} styles={styles.animatedSaveButton} />
                  </Section.Row>
                ) : (
                  <Section.Row style={styles.suffixIcon}>
                    <Icon
                      color={errors.mobile ? theme.colors.red : theme.colors.primary}
                      name="phone"
                      size={28}
                      style={styles.phoneIcon}
                    />
                  </Section.Row>
                )}
              </Section.Row>
              <ErrorText error={errors.mobile} style={styles.errorMargin} />
            </Section.Stack>
          ) : (
            <Fragment>
              {countryFlag && (
                <View style={styles.flagContainer}>
                  <SvgXml src={countryFlag} width={30} height={30} />
                </View>
              )}
              <InputRounded
                containerStyle={countryCode && styles.disabledPhoneContainer}
                disabled={true}
                error={errors.mobile}
                icon="phone"
                iconColor={theme.colors.primary}
                iconSize={28}
                placeholder="Add your Mobile"
                value={mobile ?? ''}
              />
            </Fragment>
          )}
        </Section.Row>
        <Section.Row style={!editable && styles.borderedBottomStyle}>
          <InputRounded
            disabled={!editable}
            error={errors.email}
            icon="envelope"
            iconColor={theme.colors.primary}
            iconSize={20}
            onChange={onEmailChange}
            placeholder="Add your Email"
            value={profile.email ?? ''}
            customIcon={
              <VerifyButton
                enabled={saveMode?.email}
                mode="email"
                cb={onVerifyEmail}
                styles={styles.animatedSaveButton}
              />
            }
          />
        </Section.Row>
      </KeyboardAwareScrollView>
    </Section.Row>
  )
}

const getStylesFromProps = ({ theme, errors }) => {
  return {
    borderedBottomStyle: {
      borderBottomColor: theme.colors.lightGray,
      borderBottomWidth: 1,
    },
    suffixIcon: {
      alignItems: 'center',
      display: 'flex',
      height: 40,
      justifyContent: 'center',
      position: 'absolute',
      right: 1,
      width: 32,
      zIndex: 1,
    },
    errorMargin: {
      marginTop: theme.sizes.default,
      marginBottom: theme.sizes.default,
    },
    phoneIcon: {
      paddingTop: Platform.select({
        web: 0,
        default: 5,
      }),
    },
    phoneInput: {
      borderColor: errors && (errors.mobile ? theme.colors.red : theme.colors.lightGray),
      borderRadius: 24,
      borderWidth: 1,
      color: errors && (errors.mobile ? theme.colors.red : theme.colors.text),
      position: 'relative',
      marginVertical: 4,
      paddingLeft: 10,
      paddingRight: 10,
      paddingTop: 6,
      paddingBottom: 6,
    },
    flagContainer: {
      display: 'flex',
      height: 24,
      width: 24,
      justifyContent: 'center',
      alignItems: 'center',
      overflow: 'hidden',
      borderWidth: 1,
      borderColor: theme.colors.lightGray,
      ...Platform.select({
        web: { borderRadius: '50%', borderStyle: 'solid' },
        default: { borderRadius: 24 / 2, paddingTop: 5 },
      }),
    },
    disabledPhoneContainer: {
      paddingLeft: 10,
    },
    disclaimer: {
      justifyContent: 'center',
      alignItems: 'center',
      width: '100%',
      paddingTop: 40,
      paddingBottom: 40,
      marginLeft: 'auto',
      marginRight: 'auto',
      paddingLeft: 8,
      paddingRight: 8,
    },
    animatedSaveButton: {
      position: 'absolute',
      width: getDesignRelativeWidth(110),
      height: 50,
      top: getDesignRelativeHeight(18),
      right: -10,
      marginVertical: 0,
      display: 'flex',
      justifyContent: 'flex-end',
    },
    customSuffixIcon: {
      alignItems: 'center',
      display: 'flex',
      height: '100%',
      justifyContent: 'center',
      position: 'absolute',
      right: 0,
      width: 50,
      zIndex: 1,
    },
  }
}

export default withStyles(getStylesFromProps)(ProfileDataTable)
