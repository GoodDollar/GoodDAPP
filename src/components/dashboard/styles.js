import { StyleSheet } from 'react-native'
import normalize from '../../lib/utils/normalizeText'
import { theme } from '../theme/styles'
import { getScreenHeight } from '../../lib/utils/orientation'

const isMobileHeight = getScreenHeight() < 680

export const receiveStyles = StyleSheet.create({
  wrapper: {
    justifyContent: 'flex-start',
    width: '100%',
    padding: 8,
  },
  section: {
    flex: 1,
  },
  sectionRow: {
    flexDirection: 'column',
    justifyContent: 'space-between',
    height: '100%',
  },
  qrCode: {
    marginTop: isMobileHeight ? 0 : '2rem',
    padding: '1rem',
    borderColor: '#555',
    borderWidth: 1,
    borderRadius: '4px',
  },
  addressSection: {
    marginBottom: isMobileHeight ? 0 : '1rem',
  },
  address: {
    margin: '0.5rem',
  },
  secondaryText: {
    margin: isMobileHeight ? '0.2rem' : '1rem',
    color: '#555555',
    fontSize: normalize(14),
    textTransform: 'uppercase',
  },
  lowerSecondaryText: {
    margin: '1rem',
    color: '#555555',
    fontSize: normalize(16),
  },
  headline: {
    ...theme.fontStyle,
    textTransform: 'uppercase',
    marginBottom: '1rem',
    fontSize: normalize(24),
  },
  buttonGroup: {
    width: '100%',
    flexDirection: 'row',
    marginTop: '1rem',
  },
  amountLabel: {
    ...theme.fontStyle,
    fontSize: normalize(32),
  },
  amountSymbol: {
    fontSize: normalize(12),
  },
  shareQRButton: {
    width: '100%',
  },
  doneButton: {
    marginTop: '1rem',
  },
  fullWidth: {
    marginHorizontal: 10,
  },
  amountSuffix: {
    flexGrow: 1,
    height: 40,
    fontSize: normalize(10),
    justifyContent: 'center',
    lineHeight: 40,
    paddingTop: 10,
  },
  buttonStyle: {
    marginTop: '1em',
  },
})
