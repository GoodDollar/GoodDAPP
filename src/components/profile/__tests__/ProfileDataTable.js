import React from 'react'
import renderer from 'react-test-renderer'
import { Provider as PaperProvider } from 'react-native-paper'
import ProfileDataTable from '../ProfileDataTable'
import { theme } from '../../theme/styles'

describe('ProfileDataTable', () => {
  const profile = { email: 'john@doe.com', mobile: '2222222222222' }

  it('matches snapshot', () => {
    const component = renderer.create(
      <PaperProvider theme={theme}>
        <ProfileDataTable profile={profile} />
      </PaperProvider>,
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('matches snapshot editable', () => {
    const component = renderer.create(
      <PaperProvider theme={theme}>
        <ProfileDataTable profile={profile} editable={true} />
      </PaperProvider>,
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('matches snapshot editable with errors', () => {
    const component = renderer.create(
      <PaperProvider theme={theme}>
        <ProfileDataTable profile={profile} editable={true} errors={{ mobile: 'error', email: 'error' }} />
      </PaperProvider>,
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
