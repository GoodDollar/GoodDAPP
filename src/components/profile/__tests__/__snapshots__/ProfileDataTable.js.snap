// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileDataTable matches snapshot 1`] = `
<div
  className="css-view-1dbjc4n r-flex-13awgt0 r-pointerEvents-12vffkv"
>
  <div
    className="css-view-1dbjc4n r-alignItems-1awozwy r-flexDirection-18u37iz r-justifyContent-1wtj0ep"
    style={
      Object {
        "WebkitBoxFlex": 1,
        "WebkitFlexGrow": 1,
        "flexGrow": 1,
        "msFlexPositive": 1,
      }
    }
  />
</div>
`;

exports[`ProfileDataTable matches snapshot editable 1`] = `
<div
  className="css-view-1dbjc4n r-flex-13awgt0 r-pointerEvents-12vffkv"
>
  <div
    className="css-view-1dbjc4n r-alignItems-1awozwy r-flexDirection-18u37iz r-justifyContent-1wtj0ep"
    style={
      Object {
        "WebkitBoxFlex": 1,
        "WebkitFlexGrow": 1,
        "flexGrow": 1,
        "msFlexPositive": 1,
      }
    }
  />
</div>
`;

exports[`ProfileDataTable matches snapshot editable with errors 1`] = `
<div
  className="css-view-1dbjc4n r-flex-13awgt0 r-pointerEvents-12vffkv"
>
  <div
    className="css-view-1dbjc4n r-alignItems-1awozwy r-flexDirection-18u37iz r-justifyContent-1wtj0ep"
    style={
      Object {
        "WebkitBoxFlex": 1,
        "WebkitFlexGrow": 1,
        "flexGrow": 1,
        "msFlexPositive": 1,
      }
    }
  />
</div>
`;
